### PR TITLE
Windows implemented

### PR DESCRIPTION
**Pull request by kobywchan**
Full functionality of Plugin.UWP including connect, scan, get services/descriptors/characteristics, read and write characteristics/descriptors, event handlers, accessor methods, etc. I apologize for the lack of commits, I am new to git and was working completely locally for the great majority of the project :)

Known Issues for UWP:
-Disconnect cannot be implemented with UWP API
-GetSystemConnectedOrPairedDevices cannot be implemented without being async
-GetInitialState only returns connected or disconnected (also due to UWP API limitations)
-RequestMtu is not supported in windows
-Rssi cannot be updated since Rssi information is only recieved in nativeadvertisements
-UpdateConnectionInterval is not supported in UWP
-IsPrimary returns true as default since the method to get parent services is now obselete in the UWP API

### CODE REVIEW

**Comment by smstuebe** [comment]
cool. Thanks. I think we should merge the other PR, release 1.3. and then use this for the next beta

**Comment by kobywchan** [comment]
Awesome, thanks. I'll keep my eye out for anything else I can help with!

File: Source/Plugin.BLE.UWP/Plugin.BLE.UWP.csproj [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{CCD85512-A2D7-4076-B31D-7A24829058AF}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Plugin.BLE.UWP</RootNamespace>
+    <AssemblyName>Plugin.BLE.UWP</AssemblyName>
```
Comment (smstuebe) [inline Source/Plugin.BLE.UWP/Plugin.BLE.UWP.csproj]: Has to be `Plugin.BLE` for bait and switch pattern

File: Source/Plugin.BLE.UWP/Plugin.BLE.UWP.csproj [hunk 2]
```diff
@@ -0,0 +1,147 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{CCD85512-A2D7-4076-B31D-7A24829058AF}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Plugin.BLE.UWP</RootNamespace>
```
Comment (smstuebe) [inline Source/Plugin.BLE.UWP/Plugin.BLE.UWP.csproj]: Should be `Plugin.BLE` to be same as Android project.

**Comment by xabre** [comment]
Great work @kobywchan . Can't wait to check it out :)

**Comment by xabre** [comment]
I've decided to open a new uwp branch called `uwp_creators_update` and merge the PR here. We will do all required changes/testing here and then merge it to master :)

**Comment by smstuebe** [comment]
@xabre Please merge it to 2.0.0. This is a good chance to switch to netstandard :)
Merging to master will block us from releasing minor fixes. So please don't :)

**Comment by xabre** [comment]
I will merge it to our release branches as soon as I can test it.  For now it's in `uwp_creators_update`

**Comment by flyingxu** [comment]
Any beta nuget package so that I can also test this branch?

**Comment by smstuebe** [comment]
@flyingxu we will merge that into 2.0.0 and release an alpha. But we are currently peparing our local Xamarin Dev Days. This is not our fulltime project. So please be patient.

**Comment by flyingxu** [comment]
Sure, I understand. 
My current workaround is to copy the whole Plugin.Ble.UPW folder into my own project source base and change the relative project reference to NuGet reference of Plugin.Ble. It seems working well and I don't need to change any code.

Have fun in your Xamarin Dev Days, I'm jealous of you.

**Pull request merged**